src/consumption.cpp

```diff
@@ -732,6 +732,19 @@ void player::consume_effects( item &food, bool rotten )
             fun = 1;
         }
     }
+    //Food is made out of meat (use ALLERGEN_MEAT, not CARNIVORE_OK (the latter has eggs and milk),
+    //Reduces bad morale penalties on food only made of meat.
+    if( has_trait( "CARNIVORE" ) && food.has_flag( "ALLERGEN_MEAT" ) &&
```

Comment:
`ALLERGEN_MEAT` flags things that are just dusted with meat due to processing. That is, a giant pizza with a single slice of ham will be `ALLERGEN_MEAT`.